server/src/internalClusterTest/java/org/elasticsearch/snapshots/CustomMetadataSnapshotIT.java

```diff
@@ -43,111 +41,110 @@
-            if (randomBoolean()) {
-                metadataBuilder.putCustom(SnapshottableMetadata.TYPE, new SnapshottableMetadata("after_snapshot_s"));
-            } else {
-                metadataBuilder.removeCustom(SnapshottableMetadata.TYPE);
-            }
-            metadataBuilder.putCustom(NonSnapshottableMetadata.TYPE, new NonSnapshottableMetadata("after_snapshot_ns"));
-            if (randomBoolean()) {
-                metadataBuilder.putCustom(SnapshottableGatewayMetadata.TYPE, new SnapshottableGatewayMetadata("after_snapshot_s_gw"));
-            } else {
-                metadataBuilder.removeCustom(SnapshottableGatewayMetadata.TYPE);
-            }
-            metadataBuilder.putCustom(NonSnapshottableGatewayMetadata.TYPE, new NonSnapshottableGatewayMetadata("after_snapshot_ns_gw"));
-            metadataBuilder.removeCustom(SnapshotableGatewayNoApiMetadata.TYPE);
-            builder.metadata(metadataBuilder);
-            return builder.build();
-        });
+    public void testShouldNotRestoreRepositoryMetadata() {
+        var repoPath = randomRepoPath();
 
-        logger.info("--> delete repository");
-        assertAcked(clusterAdmin().prepareDeleteRepository("test-repo"));
+        logger.info("create repository");
+        createRepository("test-repo-1", "fs", repoPath);
 
-        createRepository("test-repo-2", "fs", tempDir);
+        logger.info("create snapshot");
+        createFullSnapshot("test-repo-1", "test-snap");
+        assertThat(getSnapshot("test-repo-1", "test-snap").state(), equalTo(SnapshotState.SUCCESS));
 
-        logger.info("--> restore snapshot");
+        logger.info("delete repository");
+        assertAcked(clusterAdmin().prepareDeleteRepository("test-repo-1"));
+
+        logger.info("create another repository");
+        createRepository("test-repo-2", "fs", repoPath);
+
+        logger.info("restore snapshot");
         clusterAdmin().prepareRestoreSnapshot("test-repo-2", "test-snap")
             .setRestoreGlobalState(true)
             .setIndices("-*")
             .setWaitForCompletion(true)
             .execute()
             .actionGet();
 
-        logger.info("--> make sure old repository wasn't restored");
-        assertRequestBuilderThrows(clusterAdmin().prepareGetRepositories("test-repo"), RepositoryMissingException.class);
+        logger.info("make sure old repository wasn't restored");
+        assertRequestBuilderThrows(clusterAdmin().prepareGetRepositories("test-repo-1"), RepositoryMissingException.class);
         assertThat(clusterAdmin().prepareGetRepositories("test-repo-2").get().repositories().size(), equalTo(1));
+    }
+
+    public void testShouldRestoreOnlySnapshotMetadata() throws Exception {
+        var repoPath = randomRepoPath();
+
+        logger.info("create repository");
+        createRepository("test-repo", "fs", repoPath);
+
+        logger.info("add custom persistent metadata");
+        boolean isSnapshotMetadataSet = true;// randomBoolean();
```

Comment:
This should become random once https://github.com/elastic/elasticsearch/pull/81373 is merged

Reply:
Maybe make the comment a `TODO:` then :) ?

Reply:
I will sync and merge after that pr